extensions/amp-story/0.1/amp-story-info-dialog.js

```diff
@@ -0,0 +1,261 @@
+   * system sharing, or the fallback UI.
+   * @param {boolean} isOpen
+   * @private
+   */
+  onInfoDialogStateUpdated_(isOpen) {
+    this.vsync_.mutate(() => {
+      this.element_.classList.toggle(DIALOG_VISIBLE_CLASS, isOpen);
+    });
+  }
+
+  /**
+   * Handles click events and maybe closes the dialog.
+   * @param  {!Event} event
+   */
+  onInfoDialogClick_(event) {
+    const el = dev().assertElement(event.target);
+    // Closes the dialog if click happened outside of the dialog main container.
+    if (!closest(el, el => el === this.innerContainerEl_, this.element_)) {
+      this.close_();
+    }
+  }
+
+  /**
+   * Closes the info dialog.
+   * @private
+   */
+  close_() {
+    this.storeService_.dispatch(Action.TOGGLE_INFO_DIALOG, false);
+  }
+
+  /**
+   * @return {!Promise<?string>} The URL to visit to receive more info on this
+   *     page.
+   * @private
+   */
+  requestMoreInfoLink_() {
+    const messagingPromise = this.viewer_.whenMessagingReady();
+
+    if (!messagingPromise) {
+      // There is no viewer to supply the more info URL.
+      return Promise.resolve();
+    }
+
+    return messagingPromise
+        .then(() => {
+          return this.viewer_.sendMessageAwaitResponse('moreInfoLinkUrl',
+              /* data */ undefined);
+        })
+        .then(moreInfoUrl => {
+          if (!moreInfoUrl) {
+            return null;
+          }
+
+          return assertAbsoluteHttpOrHttpsUrl(dev().assertString(moreInfoUrl));
+        });
+  }
+
+  /**
+   * Sets the heading on the dialog.
+   */
+  setHeading_() {
+    const label = this.localizationService_.getLocalizedString(
+        LocalizedStringId.AMP_STORY_DOMAIN_DIALOG_HEADING_LABEL);
+
+    return this.vsync_.runPromise({
+      measure: state => {
+        state.headingEl = this.element_
+            .querySelector('.i-amphtml-story-info-heading');
+      },
+      mutate: state => {
+        state.headingEl.textContent = label;
+      },
+    }, /* state */ {});
+  }
+
+  /**
+   * @param {string} pageUrl The URL to the canonical version of the current
+   *     document.
+   */
+  setPageLink_(pageUrl) {
+    return this.vsync_.runPromise({
+      measure: state => {
+        state.linkEl = this.element_
+            .querySelector('.i-amphtml-story-info-link');
+      },
+      mutate: state => {
+        state.linkEl.setAttribute('href', pageUrl);
+        state.linkEl.setAttribute('rel', 'amphtml');
+
+        // Add zero-width space character (\u200B) after "." and "/" characters
+        // to help line-breaks occur more naturally.
+        state.linkEl.textContent = pageUrl.replace(/([/.]+)/gi, '$1\u200B');
+      },
+    }, /* state */ {});
+  }
+
+  /**
+   * @param {?string} moreInfoUrl The URL to the "more info" page, if there is
+   * one.
+   */
+  setMoreInfoLinkUrl_(moreInfoUrl) {
+    if (!moreInfoUrl) {
+      return Promise.resolve();
+    }
+
+    return this.vsync_.runPromise({
```

Comment:
Suggestion: We're trying to move away from vsync, and preferred method is baseElement.measureMutateElement() or using `resources` like[ seen here](https://github.com/Enriqe/amphtml/blob/3edf797df55c4cd415a2d5497f69983d55fca1d2/extensions/amp-story/1.0/bookend/amp-story-bookend.js#L536).